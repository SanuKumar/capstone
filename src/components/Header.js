import React from 'react'
import { Container, Nav, Navbar, NavDropdown, Form, Image } from 'react-bootstrap';
import { useHistory } from 'react-router-dom';
import { FiLogIn } from 'react-icons/fi';
import { BiLogOut } from 'react-icons/bi';

const Header = ({ handleProductSearch, isUserLoggedIn }) => {
  let history = useHistory()
  const UserMenu = (
    <>
      {isUserLoggedIn ?
        <>
          <Image
            src={'https://upload.wikimedia.org/wikipedia/commons/thumb/1/12/User_icon_2.svg/800px-User_icon_2.svg.png'}
            alt="UserName profile image"
            roundedCircle
            style={{ width: '40px' }}
          />
          <span style={{ color: "white" }}>
            {isUserLoggedIn.firstname}
          </span>
        </> : <span style={{ color: "white" }}>Login/Register</span>}
    </>
  )

  const handleLogout = () => {
    sessionStorage.clear();
    history.push('/')
    window.location.reload();
  }

  return (
    <Navbar bg="primary" variant="dark" expand="lg" fixed="top">
      <Container>
        <Navbar.Brand onClick={() => history.push('/')} style={{ cursor: "pointer" }}>Product Inventory</Navbar.Brand>
        <Navbar.Toggle aria-controls="basic-navbar-nav" />
        <Navbar.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <Nav.Link onClick={() => history.push("/about")}>About</Nav.Link>
            <Nav.Link onClick={() => history.push("/chart")} style={{ marginRight: "2rem" }}>Product's Chart</Nav.Link>
            <Form className="d-flex">
              <Form.Control
                type="search"
                placeholder="Search"
                className="me-2"
                aria-label="Search"
                onChange={(e) => handleProductSearch(e.target.value)}
              />
            </Form>
          </Nav>
        </Navbar.Collapse>
        <Navbar.Collapse className="justify-content-end">
          <NavDropdown
            title={UserMenu}
            id="basic-nav-dropdown"
          >
            <NavDropdown.Item onClick={() => history.push("/about")}>About</NavDropdown.Item>
            <NavDropdown.Item onClick={() => history.push('/register')}>Register <NavDropdown.Divider /></NavDropdown.Item>
            {isUserLoggedIn
              ? <NavDropdown.Item onClick={handleLogout}>Logout <BiLogOut /></NavDropdown.Item>
              : <NavDropdown.Item onClick={() => history.push('/login')}>Login <FiLogIn /></NavDropdown.Item>
            }
          </NavDropdown>
        </Navbar.Collapse>
      </Container>
    </Navbar>
  )
}

export default Header