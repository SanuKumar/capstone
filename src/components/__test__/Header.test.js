import React from "react"
import { render, cleanup } from "@testing-library/react"
import Header from "../Header"

afterEach(cleanup)

describe("Header page snapshot", () => {
  test("renders correctly", () => {
    const { asFragment } = render(<Header />);
    expect(asFragment()).toMatchSnapshot();
  })
})