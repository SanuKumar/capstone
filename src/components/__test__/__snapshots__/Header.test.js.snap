// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header page snapshot renders correctly 1`] = `
<DocumentFragment>
  <nav
    class="navbar navbar-expand-lg navbar-dark bg-primary fixed-top"
  >
    <div
      class="container"
    >
      <span
        class="navbar-brand"
        style="cursor: pointer;"
      >
        Product Inventory
      </span>
      <button
        aria-controls="basic-navbar-nav"
        aria-label="Toggle navigation"
        class="navbar-toggler collapsed"
        type="button"
      >
        <span
          class="navbar-toggler-icon"
        />
      </button>
      <div
        class="navbar-collapse collapse"
        id="basic-navbar-nav"
      >
        <div
          class="me-auto navbar-nav"
        >
          <a
            class="nav-link"
            href="#"
            role="button"
            tabindex="0"
          >
            About
          </a>
          <a
            class="nav-link"
            href="#"
            role="button"
            style="margin-right: 2rem;"
            tabindex="0"
          >
            Product's Chart
          </a>
          <form
            class="d-flex"
          >
            <input
              aria-label="Search"
              class="me-2 form-control"
              placeholder="Search"
              type="search"
            />
          </form>
        </div>
      </div>
      <div
        class="justify-content-end navbar-collapse collapse"
      >
        <div
          class="nav-item dropdown"
        >
          <a
            aria-expanded="false"
            class="dropdown-toggle nav-link"
            href="#"
            id="basic-nav-dropdown"
            role="button"
            tabindex="0"
          >
            <span
              style="color: white;"
            >
              Login/Register
            </span>
          </a>
        </div>
      </div>
    </div>
  </nav>
</DocumentFragment>
`;
